src/lib/viewers/controls/settings/Settings.tsx

```diff
@@ -34,14 +39,22 @@ export default function Settings({
         setActiveRect(undefined);
         setIsFocused(false);
         setIsOpen(false);
-    }, []);
+
+        onClose();
+    }, [onClose]);
```

Comment:
It seems like any call to Model3dViewer's `renderUI` would reset the controls, since the handler is declared as an inline function. Is that not the case? Would it be safer to create a `useCallback` helper that can both call `setIsOpen` and `onClose` or `onOpen`?

Reply:
No, that's not the case. Seems like any call to `renderUI` should probably create a new callback for `resetControls` but not necessarily call it

Reply:
Are you proposing something like:
```
const setIsOpenHelper = React.useCallback(() => {
    const nextIsOpen = !isOpen;
    setIsOpen(nextIsOpen);

    if (nextIsOpen) {
        onOpen();
    } else {
        onClose();
    }
}, [onClose, onOpen]);
```
And use that in place of `setIsOpen`?